src/rcc/mod.rs

```diff
@@ -0,0 +1,1458 @@
+        feature = "stm32f413",
+        feature = "stm32f423",
+        feature = "stm32f446",
+    ))]
+    fn i2s_clocks(&self) -> I2sClocks {
+        let i2s_apb1_ext = self.i2s_apb1_clk.is_some() && self.i2s_apb1_clk == self.i2s_ckin;
+        let i2s_apb2_ext = self.i2s_apb2_clk.is_some() && self.i2s_apb2_clk == self.i2s_ckin;
+        let pll_i2s_clk = if i2s_apb1_ext {
+            None
+        } else {
+            self.i2s_apb1_clk
+        };
+        let pll_i2s_clk2 = if i2s_apb2_ext {
+            None
+        } else {
+            self.i2s_apb2_clk
+        };
+        if pll_i2s_clk.is_some() && pll_i2s_clk2.is_some() && pll_i2s_clk != pll_i2s_clk2 {
+            panic!("only one I2S PLL frequency implemented");
+        }
+        I2sClocks {
+            i2s_apb1_ext,
+            i2s_apb2_ext,
+            pll_i2s_clk,
+        }
+    }
+
+    #[cfg(not(any(
+        feature = "stm32f412",
+        feature = "stm32f413",
+        feature = "stm32f423",
+        feature = "stm32f446",
+    )))]
+    fn i2s_clocks(&self) -> I2sClocks {
+        let i2s_ext = self.i2s_clk.is_some() && self.i2s_clk == self.i2s_ckin;
+        let pll_i2s_clk = if i2s_ext { None } else { self.i2s_clk };
+        I2sClocks {
+            i2s_ext,
+            pll_i2s_clk,
+        }
+    }
+
+    fn flash_setup(sysclk: u32) {
+        use crate::stm32::FLASH;
+
+        #[cfg(any(
+            feature = "stm32f401",
+            feature = "stm32f405",
+            feature = "stm32f407",
+            feature = "stm32f410",
+            feature = "stm32f411",
+            feature = "stm32f412",
+            feature = "stm32f415",
+            feature = "stm32f417",
+            feature = "stm32f427",
+            feature = "stm32f429",
+            feature = "stm32f437",
+            feature = "stm32f439",
+            feature = "stm32f446",
+            feature = "stm32f469",
+            feature = "stm32f479"
+        ))]
+        let flash_latency_step = 30_000_000;
+
+        #[cfg(any(feature = "stm32f413", feature = "stm32f423"))]
+        let flash_latency_step = 25_000_000;
+
+        unsafe {
+            let flash = &(*FLASH::ptr());
+            // Adjust flash wait states
+            flash.acr.modify(|_, w| {
+                w.latency().bits(((sysclk - 1) / flash_latency_step) as u8);
+                w.prften().set_bit();
+                w.icen().set_bit();
+                w.dcen().set_bit()
+            })
+        }
+    }
+
+    /// Initialises the hardware according to CFGR state returning a Clocks instance.
+    /// Panics if overclocking is attempted.
+    pub fn freeze(self) -> Clocks {
+        self.freeze_internal(false)
+    }
+
+    /// Initialises the hardware according to CFGR state returning a Clocks instance.
+    /// Allows overclocking.
+    pub unsafe fn freeze_unchecked(self) -> Clocks {
+        self.freeze_internal(true)
+    }
+
+    pub fn freeze_internal(self, unchecked: bool) -> Clocks {
```

Comment:
Was this supposed to be `pub` ?

Reply:
I do not see why it should be, but it was already pub before I started.